app/services/catalog/order_item_transition.rb

```diff
@@ -0,0 +1,50 @@
+module Catalog
+  class OrderItemTransition
+    attr_reader :order_item_id
+    attr_reader :state
+
+    def initialize(order_item_id)
+      @order_item_id = order_item_id
+    end
+
+    def process
+      @approvals = ApprovalRequest.where(:order_item_id => @order_item_id)
+
+      if approved?
+        submit_order
+        @state = "approved"
+      elsif denied?
+        mark_denied
+        @state = "denied"
+      else
+        @state = "pending"
+      end
+
+      self
+    end
+
+    private
+
+    def submit_order
+      order_item = @approvals.first.order_item
+
+      ManageIQ::API::Common::Request.with_request(order_item.context.transform_keys(&:to_sym)) do
+        Catalog::SubmitOrder.new(order_item.order_id).process
```

Comment:
@lindgrenj6 we should catch standard error here log it as an error and suppress the exception since we don't want the caller which is a background worker to end. The SubmitOrder is already catching the topology error but its re-raising the error

Reply:
Should we do a `Rails.logger.error` as well as a Progress Message?

Reply:
@lindgrenj6 both since we have an order item